Kokkos: deprecate versions older than 3.7.2

tpadioleau opened the pull request:
This PR starts to deprecate versions 3 of Kokkos, keeping only 3.7.2.

tpadioleau committed:
Kokkos: deprecate versions older than 3.7.2

JBludau reviewed (approved):
as discussed in the last wg meeting

lucbv reviewed (changes requested):
I don't see associated changes for Kokkos Kernels?

lucbv commented:
Adding on to my remark above we also do not have the nvcc_wrapper updates either

rbberger commented:
pinging the maintainers of `ecp_data_vis_sdk`  @kwryankrattiger and @svenevs. This fails the CI due to the `kokkos@:3.7.1` dependency. Can we upgrade that to `kokkos@:3.7.2`?

kwryankrattiger commented:
Sure. That package is actually deprecated and we are actively working on moving to a different approach to shipping the DAV SDK. I think the kokkos restriction was related to restrictions with CXX version and ascent which is no longer a concern.

tpadioleau committed:
Add Kokkos Kernels 3.7.2

tpadioleau committed:
Kokkos Kernels: deprecate versions older than 3.7.2

tpadioleau committed:
Kokkos nvcc wrapper: deprecate versions older than 3.7.2

tpadioleau commented:
> I don't see associated changes for Kokkos Kernels?

@lucbv You are right, this is now fixed. Note that I have added version 3.7.2 of Kokkos Kernels.

lucbv reviewed (approved):
Thanks of all the updates @tpadioleau

rbberger commented:
@tpadioleau please bump the `ecp_data_vis_sdk` kokkos requirement to 3.7.2 to make the CI pass.

tpadioleau committed:
Paraview: bump paraview to any Kokkos 3.7 patch

tpadioleau commented:
@rbberger I have updated the Paraview package. Correct me if I am mistaken, I understand that only Paraview has a direct dependency to Kokkos
```
RuntimeError: failed to concretize `ecp-data-vis-sdk+paraview+rocm+vtkm amdgpu_target=gfx908` for the following reasons:
     1. Cannot satisfy 'kokkos@:3.7.1'
        required because paraview depends on kokkos@:3.7.1+rocm amdgpu_target=gfx908 when +rocm amdgpu_target=gfx908 
```

rbberger commented:
@tpadioleau Ah, ok. Since the CI has passed. LGTM. @kwryankrattiger as ParaView maintainer, please add an approval. I'll then go ahead and merge this.

kwryankrattiger reviewed (approved):
Thanks!
